.changeset/great-crews-own.md

```diff
@@ -0,0 +1,14 @@
+---
+'@backstage/cli': minor
+---
+
+Treat files in `__tests__` and `__mocks__` directories as test files for linting
+purposes.
```

Comment:
Wanna double check that we're aligned on the purpose of this. What I'm thinking thinking is that `__mocks__` is added for the purpose of automated mocks through `jest.mock('./my-module')` with `__mocks__/my-module.ts`. The purpose of `__tests__` would be to put test utilities in those directories, that can import `devDependencies` and also be imported from tests.

I like the last `__tests__` addition if that's what you're aiming for, but I'm feeling it's too much of a conflict with the [default jest config](https://jestjs.io/docs/28.x/configuration#testmatch-arraystring). The worry here is that anyone coming from a default Jest setup might be very confused by the fact that [we don't include __tests__ as test files](https://github.com/backstage/backstage/blob/5213ee63c85c08b44bb3d41620e0ba287831b2c0/packages/cli/config/jest.js#L200).

Simplest way forward I think would be to pick a different name than `__tests__`. Was gonna suggest some different options, but feeling like `__testUtils__` was the only sane and self-explanatory one, what do you think?

Reply:
Yep, works for me! Definitely agree that this could confuse folks who assume based on the linting config that the default jest test patterns are supported. Will update to `__testUtils__`.